File: Lista 07/Exercício 01/Triangulo.cpp

#include "Triangulo.hpp"
#include <typeinfo>
#include <stdio.h>

Triangulo::Triangulo(double &base, double &altura){
	this->setBase(base);
	this->setAltura(altura);
}

void Triangulo::setBase(double &base){
	this->base = base;
}
double Triangulo::getBase(){
	return this->base;
}

void Triangulo::setAltura(double &altura){
	this->altura = altura;
}
double Triangulo::getAltura(){
	return this->altura;
}

double Triangulo::getArea(){
	return 0.5*getAltura()*getBase();
}

Triangulo::~Triangulo(){
	printf("%s \t", typeid(this).name());
}

File: TP 01/TP01 - Matheus Gonçalves/src/figura/Triangulo.java
package figura;


public class Triangulo extends Figura{
	private int base;
	private int altura;
	
	public Triangulo(int base, int altura, char conteudo){
		this.base = base;
		this.altura = altura;
		this.conteudo = conteudo;
	}
	
	public void desenhar() {
		int a = altura/2;
		int b = base/2;
		for(int i=0; i < a; i++){
			for(int j=b*-1; j < b; j++){
				if(j >= i*-1 && j <= i)
					System.out.print(conteudo);
				else
					System.out.print(" ");
			}
			System.out.println();
		}
	}
}
		
		/*
for(int i=0; i < altura; i++){
	cont++;
	j = 0;
	while(j < cont){
		System.out.print(conteudo);
		j++;
	}
	System.out.print("\n");
}
		 */